test/stdlib/Result.swift

```diff
@@ -0,0 +1,155 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import Swift
+
+let ResultTests = TestSuite("Result")
+
+fileprivate enum Err: Error, Equatable {
+  case err
+  case derr
+}
+
+fileprivate let string = "string"
+
+ResultTests.test("Construction") {
+  let result1: Result<String, String> = .success(string)
+  let result2: Result<String, String> = .failure(string)
+  let string1: String? = {
+    switch result1 {
+      case let .success(string): 
+        return string
+      case .failure: 
+        return nil
+    }
+  }()
+  let string2: String? = {
+    switch result2 {
+      case let .failure(string): 
+        return string
+      case .success: 
+        return nil
+    }
+  }()
+
+  expectEqual(string1, string2)
+}
+
+ResultTests.test("Properties") {
+  let result1: Result<String, String> = .success(string)
+  let result2: Result<String, String> = .failure(string)
+
+  expectEqual(result1.value, result2.error)
+  expectTrue(result1.isSuccess)
+  expectFalse(result2.isSuccess)
+}
+
+ResultTests.test("Throwing Initialization and Unwrapping") {
+  func notThrowing() throws -> String {
+    return string
+  }
+
+  func throwing() throws -> String {
+    throw Err.err
+  }
+    
+  let result1 = Result { try throwing() }
+  let result2 = Result { try notThrowing() }
+    
+  expectEqual(result1.error as? Err, Err.err)
+  expectEqual(result2.value, string)
+    
+  do {
+    _ = try result1.unwrapped()
+  } catch let error as Err {
+    expectEqual(error, Err.err)
+  } catch {
+    preconditionFailure("unexpected error type")
```

Comment:
Use `expectationFailure(_:)` here; that's basically what `expectEqual(_:_:)` and friends call when they fail.

Reply:
Thanks, I didn't know what that should be, or where to find something like that. I'll update with your suggestions soon.

Reply:
Only `expectationFailure` I can find is:
```swift
expectationFailure(
  _ reason: String,
  trace message: String,
  stackTrace: SourceLocStack)
```
How would you suggest I use it?

Reply:
My mistake—looks like you want `expectUnreachable()` here.

Reply:
Updated all of the `preconditionFailure` uses to `expectUnreachable` instead.